app/controllers/pages_controller.rb

```diff
@@ -0,0 +1,15 @@
+class PagesController < ApplicationController
+  before_action :find_page
+
+  def show
+    render @page
+  end
+
+  private
+
+  def find_page
+    id = params.require(:id)
+    raise ActionController::RoutingError, "Page #{id} not found" unless Gemcutter::PAGES.include?(id)
```

Comment:
also, maybe this should `inspect` the `id` so it gets escaped?

Reply:
I'm happy to remove `#{id}` interpolation at all to keep it on super safe side.